### PR TITLE
added local minio s3 and backup:mongodb script

### PR DESCRIPTION
**Pull request by ARtheboss**

### CODE REVIEW

File: backend/src/config.ts [hunk 1]
```diff
@@ -39,4 +44,9 @@ export const config: Config = {
   SESSION_SECRET: env("SESSION_SECRET"),
   GOOGLE_CLIENT_ID: env("GOOGLE_CLIENT_ID"),
   GOOGLE_CLIENT_SECRET: env("GOOGLE_CLIENT_SECRET"),
+  S3_ENDPOINT: env("S3_ENDPOINT"),
+  S3_PORT: parseInt(env("S3_PORT")),
+  S3_ACCESS_KEY_ID: env("S3_ACCESS_KEY_ID"),
+  S3_SECRET_ACCESS_KEY: env("S3_SECRET_ACCESS_KEY"),
+  S3_MONGO_BACKUP_BUCKET: env("S3_MONGO_BACKUP_BUCKET"),
```
Comment (kevinzwang) [inline backend/src/config.ts:51]: maybe put these under an s3 key like `config.mongoDB`

File: backend/src/scripts/backup-mongodb.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    client.removeObjects(config.S3_MONGO_BACKUP_BUCKET, toDelete, (e) => {
+        if (e) {
+            console.error("Unable to delete incorrect objects. Please do so manually.")
+        }
+    });
+}
+
+(async () => {
+
+    const client = new Minio.Client({
```
Comment (kevinzwang) [inline backend/src/scripts/backup-mongodb.ts:24]: is this compatible with any S3 service? we are only using minio for local testing

**Review by kevinzwang — CHANGES_REQUESTED** [review]
Looking good. I'd also like to see a restore script (with parameters for the collections to restore) so we can test this end-to-end before merge

**Comment by ARtheboss** [comment]
@kevinzwang added your requested changes

**Comment by ARtheboss** [comment]
@kevinzwang should be good to go

**Review by kevinzwang — APPROVED** [review]
LGTM. Resolve your conflicts and then pls squash and merge!

**Pull request merged**